uni/udb/udb.icn

```diff
@@ -31,9 +36,81 @@ procedure main(argv)
    write("\n   Thank you for using UDB.\n   Goodbye !!!")
 end
 
+procedure multiprog(argv)
+   case argv[1] of {
+      "-dapproxy": { Dapproxy(argv) }
+      "-adapterfactory": { Factory(argv) }
+      default: { return }
+   }
+end
+
+class Dapproxy(argv)
+
+   method run()
+      port := (if &features == "MacOS" then "127.0.0.1" else "") || ":" || (validate_args(argv) | dapproxy_usage())
+      every trap("SIGINT" | "SIGHUP" | "SIGPIPE", dapproxyExit)
+      every !5 do
+         if dapproxySock := open(port, "n") then
+            break
+         else
+            delay(1000)
+      \dapproxySock | stop("failed to connect to ",port, " ", &errortext )
+      repeat every s := !select([dapproxySock, &input]) do
+         writes(ready(s === dapproxySock)) | writes(dapproxySock, ready())
+   end
+
+   method dapproxy_usage()
```

Comment:
we should combine all usage functions into one so that `udb -h` shows the help for all use cases in the same way. It is one application after all. Maybe we should use `options` and parse all args in one place in udb.icn, and pass the finished args to the various classes, instead of passing argv and having each class process it is own. We can also follow the gnu linux convention  of having `--` used with full word options, and `-` used for short form options. Eg, `-h` and `--help`, `-a` and `--adapter`, etc.